pandas/core/series.py

```diff
@@ -4691,8 +4731,16 @@ def rename(
             axis = self._get_axis_number(axis)
 
         if callable(index) or is_dict_like(index):
+            # error: Argument 1 to "_rename" of "NDFrame" has incompatible
+            # type "Union[Union[Mapping[Any, Hashable], Callable[[Any],
+            # Hashable]], Hashable, None]"; expected "Union[Mapping[Any,
+            # Hashable], Callable[[Any], Hashable], None]"
```

Comment:
it appears that Series.rename and DataFrame.rename have their own docstrings. So I suspect the docstring in NDFrame._rename maybe a leftover (maybe should confirm this).

In the docstring for NDFrame, the allowable types for the substitution appear to suggest that a Scalar (Hashable) is allowed. I assume that at one time the docstring was maybe a template for Series.rename.

The logic for dealing with a Scalar is only here in Series now?

Maybe we can delete that docstring in this PR to tidy things up? Since the docstring and type annotations don't match and this error suggests that Hashable should be added to NDFrame._rename. which presumable is not the case from the condition on this block and the types are not narrowed because of the is_* function

I don't recall if i have experimented with TypeGuard on `is_dict_like`, but that maybe a simpler one since only needs to narrow to Mapping?

Reply:
Will look into the doc-string later today.

About the TypeGuard: `is_dict_like` checks for only a few methods required by Mapping (I wouldn't be surprised if `is_dict_like(Series)` returns True).

Reply:
> The logic for dealing with a Scalar is only here in Series now?

Yes

> Maybe we can delete that docstring in this PR to tidy things up?

The doc-string seems to suggest that it is/was intended to be copied to sub-classes (it contains `{klass}`) but I don't see where it is used.

Do you mean to remove the entire doc-string or just the scalar comment and the examples?

Reply:
probably ok to just remove the whole docstring. (DRY)

Reply:
I removed the doc-string